### ISSUE TITLE
想问下第一章softmax回归的一个问题

### ISSUE THREAD

**Issue by yyf941126** [hzy46/Deep-Learning-21-Examples#7]
softmax.py定义交叉熵的那行里面
cross_entropy = tf.reduce_mean(-tf.reduce_sum((y_ * tf.log(y))))
tf.reduce_sum函数在这里不用指明axis吗，不指明返回的是不是标量总和，怎么mean？
对python不太熟，麻烦指点一下

**Comment by Wang-yaole** [comment]
tensorflow的文档是说If `axis` has no entries, all dimensions are reduced, and a
tensor with a single element is returned.
看来reduce_sum以后已经是是个标量了……加不加mean也就无所谓了反正也就是差个常数

**Comment by hzy46** [comment]
这个地方确实是有一些问题，实际应当是axis=1的，当时考虑为了便于理解去掉了这个参数（捂脸），就像@Wang-yaole 说的，如果加上的话就是两者之间差了一个常数

**Comment by yyf941126** [comment]
soga。。。而且居然加了axis=1之后正确率还降低了。。只有87左右

**Comment by hzy46** [comment]
需要改一下学习率的
